egress/dns-proxy/egress-dns-proxy.sh

```diff
@@ -41,7 +41,6 @@ function generate_haproxy_defaults() {
 global
     log         127.0.0.1 local2
 
-    chroot      /var/lib/haproxy
```

Comment:
Running chroot does require elevated privileges. Concretely, it requires `CAP_SYS_CHROOT` at the very least as per `man 7 capabilities`.

I didn't capture the exact error. I'll share it shortly.

However, it just doesn't make much sense to chroot when the container itself is a chroot already. Best example of that is that the [OpenShift router template](https://github.com/openshift/router/blob/master/images/router/haproxy/conf/haproxy-config.template) does not use the `chroot` setting.

Reply:
> Concretely, it requires `CAP_SYS_CHROOT`

yes, I realise this now. Apologies for the churn.

Reply:
NP. 

Just FYI, relevant error message is:
```
[ALERT] 135/094434 (1) : [haproxy.main()] Cannot chroot(/var/lib/haproxy).
```

Manually adding the `SYS_CHROOT` capability might work as a workaround as well. But it would make more sense IMHO to just drop the `chroot` setting, rely on container isolation and require as less privileges as possible.

Reply:
Indeed, I managed to reproduce `Cannot chroot(/var/lib/haproxy)` error on an unprivileged egress router container in the dns mode. Also, I confirmed that without `chroot` directive in the HAProxy config the router can be unprivileged. Interesting that the http mode (implemented using Squid proxy) [doesn't need the privileged container](https://docs.openshift.com/container-platform/4.12/networking/openshift_sdn/deploying-egress-router-http-redirection.html#nw-egress-router-pod_deploying-egress-router-http-redirection) for the router. So, this change may align the two egress router modes in that aspect.

Overall, I'm fine with the argument that `chroot` is redundant. However I would like to double check for the history of the dns egress router to make sure there was no particular reason for the `chroot` config stanza.

Reply:
I did a quick check and I found none, but worth double-checking. The `chroot` setting is in the default haproxy config file shipped with RHEL, so it is very likely it just leaked from there.